front-end-tools/src/components/DeployComponent.tsx

```diff
@@ -160,7 +143,7 @@ export default function DeployComponenet(props: ConnectionProps) {
 
                                 setBase64Module(module);
                                 setModuleReferenceCalculated(
-                                    Buffer.from(sha256([new Uint8Array(arrayBuffer)])).toString('hex')
+                                    ModuleReference.fromBuffer(Buffer.from(sha256([new Uint8Array(arrayBuffer)])))
```

Comment:
Should be able to reduce this to

```suggestion
                                    ModuleReference.fromBuffer(sha256([new Uint8Array(arrayBuffer)]))
```